src/main/java/duke/commands/ByeCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke.commands;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+/**
+ * The class for the Bye command which extends Command class.
+ */
+public class ByeCommand extends Command {
+    private String input;
+
+    /**
+     * ByeCommand constructor.
+     *
+     * @param input The user's input.
+     */
+    public ByeCommand(String input) {
+        this.input = input;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) {
+        storage.saveTaskList(tasks);
+        System.out.println("    " + "Bye. Hope to see you again soon!");
+        return true;
+    }
```

Comment:
Maybe you can pass the response text to the Ui object for them to settle instead!